fix agg split_size parameter, add docs and test (#4627)
quickwit-oss/quickwit: docs/reference/aggregation.md

> ###### **split_size**

Reviewer: ```suggestion
###### **shard_size**
```

> ###### **split_size**

Reviewer: ```suggestion
###### **shard_size**
```

Reviewer: Maybe we should only call it `shard_size`?

Author: Yes `split_size` is not a good name. It's supposed to control the amount of data that's sent between nodes, so `node_size` probably? Plus an alias to `shard_size` since it's close enough.

> The get more accurate results, we fetch more than size from each segment/split. Aliases to `shard_size`.

Reviewer: ```suggestion
The get more accurate results, we fetch more than size from each segment/split.
```

> The get more accurate results, we fetch more than size from each segment/split. Aliases to `shard_size`.

Reviewer: ```suggestion
The get more accurate results, we fetch more than size from each segment/split. 
The parameter is called `shard_size` for opensearch/elasticsearch compatibility reasons, and admits `split_size` as an alias.
```
